### ISSUE TITLE
type 'FusedTransformer' is not a subtype of type 'SyncTransformer' in type cast

### ISSUE THREAD

**Issue by vaneath** [mofalabs/sui#8]
I am using sui in flutter the latest version  **sui: ^0.2.1**. I am getting error said:

```
type 'FusedTransformer' is not a subtype of type 'SyncTransformer' in type cast
```

I'm not sure what happen, it throws this error when I use moveCall or getObject:

### Move call

```dart
  Future<void> voteCandidate() async {
    SignTransactionService signTxService = SignTransactionService();

    final tx = TransactionBlock();

    try {
      tx.moveCall(
        '${signTxService.packageObjectId}::vote::new_ballot',
        arguments: [tx.pure(candidate?.suiEventId), tx.pure(candidate?.suiCandidateId), tx.pure("This is voter hash"), tx.pure(candidate?.name)],
      );

      await signTxService.client.signAndExecuteTransactionBlock(
        signTxService.account,
        tx,
        responseOptions: SuiTransactionBlockResponseOptions(
          showEffects: true,
          showBalanceChanges: true,
          showInput: true,
          showObjectChanges: true,
        ),
        requestType: ExecuteTransaction.WaitForLocalExecution,
      );

      // Notify listeners only if the vote is successful
      notifyListeners();
    } catch (e) {
      if (kDebugMode) {
        print("Error voting candidate: $e");
      }

      // Re-throw the error to handle it in the caller function
      rethrow;
    }
  }
  ```
  
  ### Get object
  
  ```dart
    Stream<String?> getLiveEventResult() async* {
    try {
      final client = SuiClient(SuiUrls.testnet);

      _suiEventDetail = await client.getObject(event?.suiEventId ?? '', options: SuiObjectDataOptions(showContent: true));
    } catch (e) {
      if (kDebugMode) {
        print("Error getting event: $e");
      }
    }
  }
  ```

**Comment by 0xmove** [comment]
Could you provide a runnable code? I couldn’t reproduce the issue.

**Comment by dsk7990** [comment]
@vaneath can you please provide the solution for this issue?

**Comment by vaneath** [comment]
> @vaneath can you please provide the solution for this issue?

The problem is the default transformer is SyncTranformer from sui package using dio.

![IMG_20240918_071704_529.jpg](https://github.com/user-attachments/assets/6936c426-3283-49cb-b768-3a85e5f887a1)

Sorry for low quality image.

**Comment by dsk7990** [comment]
okay, then how you resolved this issue?

**Comment by 0xmove** [comment]
This is fix in v0.3.1. Dio replace the default transformer type to SyncTransformer.

**Comment by dsk7990** [comment]
Thank you @0xmove
